src/libraries/System.Private.CoreLib/src/System/Runtime/Versioning/PlatformAttributes.cs

```diff
@@ -103,4 +103,56 @@ public UnsupportedOSPlatformAttribute(string platformName) : base(platformName)
         {
         }
     }
+
+    /// <summary>
+    /// Annotates a custom guard field, property or method with a supported platform name and optional version.
+    /// Multiple attributes can be applied to indicate guard for multiple supported platforms.
+    /// </summary>
+    /// <remarks>
+    /// Callers can apply a <see cref="System.Runtime.Versioning.SupportedOSPlatformGuardAttribute " /> to a field, property or method
+    /// and use that field, property or method in a conditional or assert statements in order to safely call platform specific APIs.
+    ///
+    /// The type of the field or property should be boolean, the method return type should be boolean in order to be used as platform guard.
+    /// </remarks>
+    [AttributeUsage(AttributeTargets.Field |
+                    AttributeTargets.Method |
+                    AttributeTargets.Property,
+                    AllowMultiple = true, Inherited = false)]
+#if SYSTEM_PRIVATE_CORELIB
```

Comment:
Why is the ifdef needed?

Reply:
The is because we want to add/use the attributes on down-level TFMs where the new attributes in CoreLib are not reachable. That way, other libraries that need the attribute and still target platforms like netstandard2.0, can use these attributes by adding `<IncludePlatformAttributes>true</IncludePlatformAttributes>` property. 

This is explained more in the build rule:

https://github.com/dotnet/runtime/blob/ab84a8cfaf41327705e35d63e77ba8e34ffdc4c1/src/libraries/Directory.Build.targets#L232-L243